### PR TITLE
[PREVIEW] Disable swagger on prod

### PR DESCRIPTION
**Pull request by qzhou-hmcts**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

* Hardware:
* O/S and version:
* JDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: infrastructure/preview.tfvars [hunk 1]
```diff
@@ -1,3 +1,5 @@
 vault_env = "preprod"
 idam_api_baseurl = "https://idam-api.aat.platform.hmcts.net"
 health_check_ttl = 30000
+
+documentation_swagger_enabled="false"
```
Comment (yarpo) [inline infrastructure/preview.tfvars]: whitespaces before and after `=` please :)

File: infrastructure/prod.tfvars [hunk 2]
```diff
@@ -2,3 +2,5 @@ vault_env = "prod"
 idam_api_baseurl = "https://idam-api.platform.hmcts.net"
 
 capacity = "2"
+
+documentation_swagger_enabled="false"
```
Comment (yarpo) [inline infrastructure/prod.tfvars]: as above

File: infrastructure/preview.tfvars (hunk 1 above)
Comment (qzhou-hmcts) [reply infrastructure/preview.tfvars]: Deleted tis change as was only to test the flag

File: infrastructure/prod.tfvars (hunk 2 above)
Comment (qzhou-hmcts) [reply infrastructure/prod.tfvars]: Done

**Comment by hosainnet** [comment]
Sorry being nosy, why are we disabling it?

**Comment by qzhou-hmcts** [comment]
> Sorry being nosy, why are we disabling it?

Disabling on production to increase security as swagger is a development tool

**Comment by hosainnet** [comment]
> > Sorry being nosy, why are we disabling it?
> 
> Disabling on production to increase security as swagger is a development tool

What security risk? They're visible on AAT (unless you mean they're exposed on the .com domain?

**Comment by hosainnet** [comment]
> > > Sorry being nosy, why are we disabling it?
> > 
> > 
> > Disabling on production to increase security as swagger is a development tool
> 
> What security risk? They're visible on AAT (unless you mean they're exposed on the .com domain?

Actually that doesn't make sense, because the BE apps are not external outside the organisation - I think it's _good_ to keep swagger on in prod, at least I don't see a good reason to disable it

**Comment by qzhou-hmcts** [comment]
> > > > Sorry being nosy, why are we disabling it?
> > > 
> > > 
> > > Disabling on production to increase security as swagger is a development tool
> > 
> > 
> > What security risk? They're visible on AAT (unless you mean they're exposed on the .com domain?
> 
> Actually that doesn't make sense, because the BE apps are not external outside the organisation - I think it's _good_ to keep swagger on in prod, at least I don't see a good reason to disable it

Why do you want to keep swagger on prod?

**Comment by hosainnet** [comment]
> > > > > Sorry being nosy, why are we disabling it?
> > > > 
> > > > 
> > > > Disabling on production to increase security as swagger is a development tool
> > > 
> > > 
> > > What security risk? They're visible on AAT (unless you mean they're exposed on the .com domain?
> > 
> > 
> > Actually that doesn't make sense, because the BE apps are not external outside the organisation - I think it's _good_ to keep swagger on in prod, at least I don't see a good reason to disable it
> 
> Why do you want to keep swagger on prod?

Why not? I read the Jira and it has zero reason for this change. It has zero harm - btw, our code is already visible to every person around the world :)

**Comment by qzhou-hmcts** [comment]
> > > > > > Sorry being nosy, why are we disabling it?
> > > > > 
> > > > > 
> > > > > Disabling on production to increase security as swagger is a development tool
> > > > 
> > > > 
> > > > What security risk? They're visible on AAT (unless you mean they're exposed on the .com domain?
> > > 
> > > 
> > > Actually that doesn't make sense, because the BE apps are not external outside the organisation - I think it's _good_ to keep swagger on in prod, at least I don't see a good reason to disable it
> > 
> > 
> > Why do you want to keep swagger on prod?
> 
> Why not? I read the Jira and it has zero reason for this change. It has zero harm - btw, our code is already visible to every person around the world :)

Because we are not a public API and we use swagger to make easy the development. Unless you want to develop in prod I don't see reason to have it

**Comment by vianvi** [comment]
one reason is not to play with PROD data, if we enable swagger, there are few end points - which doesn't require Authentication/ less secure - if we trigger those events- it might cause problems to PROD Data

**Comment by hosainnet** [comment]
> one reason is not to play with PROD data, if we enable swagger, there are few end points - which doesn't require Authentication/ less secure - if we trigger those events- it might cause problems to PROD Data

This is "Security through obscurity" - if someone really wanted to do that they can use AAT swagger/postman to do the same ;)

**Pull request merged**